mycore-solr/src/main/java/org/mycore/solr/auth/MCRSolrAuthenticationFactory.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.solr.auth;
+
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.solr.client.solrj.SolrRequest;
+import org.mycore.common.config.MCRConfiguration2;
+import org.mycore.solr.MCRSolrConstants;
+
+/**
+ * Interface for adding authentication to Solr requests.
+ */
+public interface MCRSolrAuthenticationFactory {
```

Comment:
Is `...Factory` a good name here? The interface doesn't really behave like any other factory in the code base (that i could find) or how I'd expect a factory to behave.

Reply:
I could rename that to `MCRSolrAuthenticationManager`, any better ideas?